### PR TITLE
Allow building images from git worktrees

### LINKED ISSUE #2052: Container image builds don't work with git worktrees
When you set up a git worktree the `.git` folder is something like a "symlink" to the original directory
```
git worktree list
/home/tnozicka/dev/go/src/github.com/scylladb/scylla-operator    db74ccf1b [fix-openshift]
/home/tnozicka/dev/go/src/github.com/scylladb/scylla-operator-1  32f8fa4d6 [git-worktrees]
```
```
$ cat /home/tnozicka/dev/go/src/github.com/scylladb/scylla-operator/.git
cat: /home/tnozicka/dev/go/src/github.com/scylladb/scylla-operator/.git: Is a directory

$ cat /home/tnozicka/dev/go/src/github.com/scylladb/scylla-operator-1/.git
gitdir: /home/tnozicka/dev/go/src/github.com/scylladb/scylla-operator/.git/worktrees/scylla-operator-1
```

Without containerization that works fine but container image builds require a build context and the tar only the local "scylla-operator-1" directory which contains only the `.git` "symlinks" which lacks the source in the build container.

We should allow builds from source code without git in general and fix this as well.

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
This PR allows building container images without git an/or with git worktrees.

**Which issue is resolved by this Pull Request:**
Resolves #2052

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,10 +9,13 @@ IMAGE_REF ?= docker.io/scylladb/scylla-operator:$(IMAGE_TAG)
 
 MAKE_REQUIRED_MIN_VERSION:=4.2 # for SHELLSTATUS
 
-GIT_TAG ?=$(shell git describe --long --tags --abbrev=7 --match 'v[0-9]*')$(if $(filter $(.SHELLSTATUS),0),,$(error git describe failed))
-GIT_TAG_SHORT ?=$(shell git describe --tags --abbrev=7 --match 'v[0-9]*')$(if $(filter $(.SHELLSTATUS),0),,$(error git describe failed))
-GIT_COMMIT ?=$(shell git rev-parse --short "HEAD^{commit}" 2>/dev/null)$(if $(filter $(.SHELLSTATUS),0),,$(error git rev-parse failed))
-GIT_TREE_STATE ?=$(shell ( ( [ ! -d ".git/" ] || git diff --quiet ) && echo 'clean' ) || echo 'dirty')
+# Support container builds git worktree where the parent git folder isn't available
```
Comment (rzetelskik) [inline Makefile:12]:
```suggestion
# Support container builds from git worktrees where the parent git folder isn't available
```

File: Makefile [hunk 2]
```diff
@@ -9,10 +9,13 @@ IMAGE_REF ?= docker.io/scylladb/scylla-operator:$(IMAGE_TAG)
...
+GIT ?=git
+
+GIT_TAG ?=$(shell [ ! -d ".git/" ] || $(GIT) describe --long --tags --abbrev=7 --match 'v[0-9]*')$(if $(filter $(.SHELLSTATUS),0),,$(error $(GIT) describe failed))
+GIT_TAG_SHORT ?=$(shell [ ! -d ".git/" ] || $(GIT) describe --tags --abbrev=7 --match 'v[0-9]*')$(if $(filter $(.SHELLSTATUS),0),,$(error $(GIT) describe failed))
+GIT_COMMIT ?=$(shell [ ! -d ".git/" ] || $(GIT) rev-parse --short "HEAD^{commit}" 2>/dev/null)$(if $(filter $(.SHELLSTATUS),0),,$(error $(GIT) rev-parse failed))
```
Comment (rzetelskik) [inline Makefile:17]: this way you'll end up with an empty version string from `scylla-operator version` if you build it without `.git`, is this expected?

Comment (tnozicka) [reply Makefile:17]: It is. If you don't have git you don't have the version and you get the default.

File: Makefile (hunk 1 above)
Comment (tnozicka) [reply Makefile]: lost `with` somewhere. fixed

**Review by rzetelskik — APPROVED** [review]
lgtm
/assign zimnx

**Review by zimnx — APPROVED** [review]
/lgtm

**Pull request merged**